tailwind.config.js

```diff
@@ -0,0 +1,221 @@
+// tailwind.config.js
+
+import tailwindcssAnimate from "tailwindcss-animate";
+import { colorOpacityPlugin } from "./src/lib/plugins/colorOpacityPlugin";
+import colors from "tailwindcss/colors";
+
+/** @type {import("tailwindcss").Config} */
+module.exports = {
+  content: ["./src/**/*.{js,ts,jsx,tsx}", "./src/styles/tailwind.css"],
+  darkMode: ["class"],
+  plugins: [tailwindcssAnimate, colorOpacityPlugin],
+  corePlugins: {
+    preflight: false,
+  },
+  // https://github.com/tailwindlabs/tailwindcss/blob/main/stubs/config.full.js
+  theme: {
+    colors: {
+      inherit: colors.inherit,
+      current: colors.current,
+      transparent: colors.transparent,
+      black: colors.black,
+      red: colors.red,
+      yellow: colors.yellow,
+      // slate: colors.slate,
+      // gray: colors.gray,
+      // zinc: colors.zinc,
+      // neutral: colors.neutral,
+      // stone: colors.stone,
+      // orange: colors.orange,
+      // amber: colors.amber,
+      // lime: colors.lime,
+      // green: colors.green,
+      // emerald: colors.emerald,
+      // teal: colors.teal,
+      // cyan: colors.cyan,
+      // sky: colors.sky,
+      // blue: colors.blue,
+      // indigo: colors.indigo,
+      // violet: colors.violet,
+      // purple: colors.purple,
+      // fuchsia: colors.fuchsia,
+      // pink: colors.pink,
+      // rose: colors.rose,
+
+      // preDefine CSS variables in Obsidian.(https://docs.obsidian.md/Reference/CSS+variables/Foundations/Colors)
+      base: {
+        "00": "var(--color-base-00)",
+        "05": "var(--color-base-05)",
+        10: "var(--color-base-10)",
+        20: "var(--color-base-20)",
+        25: "var(--color-base-25)",
+        30: "var(--color-base-30)",
+        35: "var(--color-base-35)",
+        40: "var(--color-base-40)",
+        50: "var(--color-base-50)",
+        60: "var(--color-base-60)",
+        70: "var(--color-base-70)",
+        100: "var(--color-base-100)",
+      },
+      // red: "var(--color-red)",
```

Comment:
can we not use the obsidian defined red and yellow color?

Reply:
The defalut color offers more choices. Why we need the red color?

![CleanShot 2024-12-26 at 14 21 29](https://github.com/user-attachments/assets/094cc9b5-37f2-4c4f-a4b5-db8c347aad9d)

Reply:
This is the principle of limiting the choice for developers and consistency. 

Regarding limiting the choice, TailwindCSS is a generic CSS library; its goal is to allow people to build any applications. TW has to provide not only a complete set of primitive tokens and semantic tokens and let people decide what they want. Our goal is not to replicate TailwindCSS tokens but to tailor them down to what the copilot plugin may need. We don't want developers to add tokens like `bg-color-red-50` without a specific reason in the plugin because #fef2f2 is not a color in either the obsidian design system or copilot design.

For consistency, this PR defines red, yellow, and black differently from the rest of the colors, making it harder to predict when applying color tokens. We would either go with Tailwind color tokens entirely or choose only obsidian colors. Staying in the middle will make decision-making harder in the future.

Unless we have a concrete use case that the obsidian color tokens cannot fit and it adds value to the copilot design, we should refrain from introducing new tokens.

Reply:
> Staying in the middle will make decision-making harder in the future.

agree.

But Why we need red color? Isn't it a primitive color?

Reply:
below pic -> text: yellow-700, bg: yellow-50, What should I do? 

![Google Chrome 2024-12-26 16 09 28](https://github.com/user-attachments/assets/d17494a6-eb15-4361-b838-e08cc6c07b2e)

Reply:
This is the callout component used in the obsidian app and site. 
![image](https://github.com/user-attachments/assets/9c18dc3e-88a4-4c50-a4b6-69abe7d27e14)


It uses --callout-warning which is --color-orange-rgb. The background is rgba(var(--callout-warning), 0.1)). There is no specific token for it. 

My obsidian tailwind project didn't include all the component tokens from obsidian yet. You can add this token to your PR. Then you can build it as `bg-callout-warning/10` and `text-callout-warning`. Here's the reference https://docs.obsidian.md/Reference/CSS+variables/Editor/Callout